code/datums/traits/negative.dm

```diff
@@ -370,3 +370,108 @@
 		if(prob(1))
 			new/obj/item/reagent_containers/food/snacks/spaghetti/pastatomato(get_turf(H)) //now that's what I call spaghetti code
 
+//If you want to make some kind of junkie variant, just extend this quirk.
+/datum/quirk/junkie
+	name = "Junkie"
+	desc = "You can't get enough of hard drugs."
+	value = -2
+	var/drug_list = list("crank", "krokodil", "morphine", "happiness", "methamphetamine") //List of possible IDs
+	var/reagent_id //ID picked from list
+	var/datum/reagent/reagent_type //If this is defined, reagent_id will be unused and the defined reagent type will be instead.
+	var/datum/reagent/R
+	gain_text = "<span class='danger'>You suddenly feel the craving for drugs.</span>"
+	lose_text = "<span class='notice'>You feel like you should kick your drug habit.</span>"
+	medical_record_text = "Patient has a history of hard drugs."
+	var/where_drug
+	var/obj/item/drug_container_type //If this is defined before pill generation will be skipped, this is the type of the pill bottle.
+	var/obj/item/D
+	var/where_accessory
+	var/obj/item/accessory_type //If this is null, it won't be spawned.
+	var/obj/item/A
+	var/tick_counter = 0
+
+/datum/quirk/junkie/on_spawn()
+	var/mob/living/carbon/human/H = quirk_holder
+	reagent_id = pick(drug_list)
+	if (!reagent_type)
+		reagent_type = GLOB.chemical_reagents_list[reagent_id]
+	R = new reagent_type(null)
+	H.reagents.addiction_list.Add(R)
+	if (!drug_container_type)
+		drug_container_type = /obj/item/storage/pill_bottle
+		D = new drug_container_type(get_turf(quirk_holder))
```

Comment:
Instead of duplicating this line, you could do something like

```dm
if(!drug_container_type)
	drug_container_type = /obj/item/storage/pill_bottle
D = new drug_container_type(...)
if(istype(D, /obj/item/storage/pill_bottle))
	...
```